app/upload.js

```diff
@@ -1,118 +1,159 @@
-			o += String.fromCharCode.apply(null, new Uint16Array(data.slice(l * w)));
-			return o;
-		}
-
-		function s2ab(s) {
-			var b = new ArrayBuffer(s.length * 2),
-			v = new Uint16Array(b);
-			for (var i = 0; i != s.length; ++i) {
-				v[i] = s.charCodeAt(i);
-			}
-			return [v, b];
-		}
-
-		function xw_xfer(data, cb) {
-			var worker = new Worker(XW.rABS);
-			worker.onmessage = function (e) {
-				switch (e.data.t) {
-				case 'ready':
-					break;
-				case 'e':
-					console.error(e.data.d);
-					break;
-				default:
-					xx = ab2str(e.data).replace(/\n/g, "\\n").replace(/\r/g, "\\r");
-					console.log("done");
-					cb(JSON.parse(xx));
-					break;
+define(['app/database', 'jquery', 'bootstrap', 'xlsx', 'parsley', 'bootstrap-select'], function (db, $, bootstrap, XLSX, parsley, select) {
+	var X = XLSX;
+	var saveData;
+	var correctUpload = false;
+
+	// Script for evaluating the input of the upload form
+	$(function () {
+		$(".selectpicker").selectpicker();
+		window.Parsley.on('field:error', function() {
+				if (this.$element.is("select")) {
+						this.$element.parent().children('.selectpicker').selectpicker('setStyle', 'alert-danger').selectpicker('refresh');
 				}
-			};
-			var val = s2ab(data);
-			worker.postMessage(val[1], [val[1]]);
-
-		}
-
-		function to_json(workbook) {
-			var result = {};
-			workbook.SheetNames.forEach(function (sheetName) {
-				var roa = XLSX.utils.sheet_to_row_object_array(workbook.Sheets[sheetName]);
-				if (roa.length > 0) {
-					result[sheetName] = roa;
+		});
+		window.Parsley.on('field:success', function() {
+				if (this.$element.is("select")) {
+						this.$element.parent().children('.selectpicker').selectpicker('setStyle', 'alert-success', 'add').selectpicker('setStyle', 'alert-danger', 'remove').selectpicker('refresh');
 				}
-			});
+		});
+		$('select').on('changed.bs.select', function (e) {
+			$(this).selectpicker('setStyle', 'alert-success', 'add').selectpicker('setStyle', 'alert-danger', 'remove').selectpicker('refresh');
+			$(this).parent().children(".parsley-errors-list").html("");
+		});
+		$('select').on('rendered.bs.select', function (e) {
+			$(this).parent().removeClass("parsley-error");
+		});
+
+		// Initiate form error handling
+		$('#uploadForm').parsley().on('field:validated', function() {
+			var ok = $('.parsley-error').length === 0;
+			$('.bs-callout-info').toggleClass('hidden', !ok);
+			$('.bs-callout-warning').toggleClass('hidden', ok);
+		})
+		.on('form:submit', function() {
+			return false; // Don't submit form
+		})
+		.on('form:success', function() {
+			console.log("done");
+			process_wb(JSON.parse(saveData));
+			window.location = 'index.html';
+		});
+	});
 
-			return result;
+	window.Parsley.addValidator('fileXlsx', {
+		validateString: function(_value, maxSize, parsleyInstance) {
+			return correctUpload;
+		},
+		requirementType: 'integer',
+		messages: {
+			en: 'This file is not supported.',
+			fr: "Ce fichier est plus grand que %s Kb."
```

Comment:
The french and english error message have a completely different meaning. If the French message is correct the translation is kind of misleading.